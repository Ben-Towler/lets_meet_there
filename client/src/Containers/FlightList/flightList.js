import React from 'react';
import './flightList.css';
import FlightTile from '../../Components/FlightTile/flightTile';

export default ({ matchedFlights, places, carriers }) => {
  console.log(matchedFlights);
  return (
    <div>
      <div className="tile_Container">
        {Object.keys(matchedFlights).map((key) => (
          <FlightTile
            flight1={matchedFlights[key][0][0][0]} //extra []  here sort it out
            flight2={matchedFlights[key][1][0][0]}
            location={key}
            places={places}
            carriers={carriers}
          ></FlightTile>
        ))}
      </div>
    </div>
  );
};
