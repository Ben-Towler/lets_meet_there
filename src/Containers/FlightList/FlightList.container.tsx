import React from 'react';
import { Trip } from 'Components';

interface Props {
  matchedFlights?: any;
  user?: any;
}

function FlightList ({ matchedFlights, user }: Props): JSX.Element {
  return (
    <div>
      <div className="tile_Container">
        {matchedFlights.map((flight) => (
          <React.Fragment key={flight[0] + 1}>
            <Trip
              user={user}
              key={flight[0]}
              yourFlight={flight[1][0][0]}
              friendsFlight={flight[2][0][0]}
              location={flight[0]}
            />
          </React.Fragment>
        ))}
      </div>
    </div>
  );
};

export default FlightList;