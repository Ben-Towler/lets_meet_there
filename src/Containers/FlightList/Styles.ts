import styled from 'styled-components';

export const StyledFlightList = styled.div`
  max-width: 70%;
`;