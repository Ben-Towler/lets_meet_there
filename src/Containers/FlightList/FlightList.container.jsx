import React from 'react';
import './flightList.css';
import { Trip } from '../../Components';

function FlightList ({ matchedFlights }) {
  return (
    <div>
      <div className="tile_Container">
        {matchedFlights.map((flight) => (
          <React.Fragment>
            <Trip
              key={flight[0]}
              yourFlight={flight[1][0][0]}
              friendsFlight={flight[2][0][0]}
              location={flight[0]}
            />
          </React.Fragment>
        ))}
      </div>
    </div>
  );
};

export default FlightList;