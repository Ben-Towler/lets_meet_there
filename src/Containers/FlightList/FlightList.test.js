import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'
import { configure } from 'enzyme';
import FlightList from './flightList';
import Adapter from 'enzyme-adapter-react-16';
import { flightListMock } from '../../Mocks/data.mock';
configure({adapter: new Adapter()});

describe('Flight Tile', () => {

  it('Matches snapshot', () => {
    const renderer = new ShallowRenderer()
    const result = renderer.render(
      <FlightList matchedFlights={flightListMock} />
    );
    expect(result).toMatchSnapshot()
  })

});