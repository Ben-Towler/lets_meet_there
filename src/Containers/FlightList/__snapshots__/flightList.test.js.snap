// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Tile Matches snapshot 1`] = `
<div>
  <div
    className="tile_Container"
  >
    <React.Fragment>
      <Trip
        location={66328}
      />
    </React.Fragment>
  </div>
</div>
`;
